packages/components/progress-bar/src/progress-bar.ts

```diff
@@ -0,0 +1,98 @@
+import { localized, msg } from '@lit/localize';
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { Icon } from '@sl-design-system/icon';
+import { type CSSResultGroup, LitElement, type TemplateResult, html, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import styles from './progress-bar.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-progress-bar': ProgressBar;
+  }
+}
+
+export type ProgressState = 'active' | 'success' | 'warning' | 'error';
+
+/**
+ * Progress bar component that can be used to communicate process status.
+ * Hidden aria-live element will help making the progress bar more accessible and should cause fewer bugs in products than leaving it up for the developers.
+ *
+ *
+ * ```html
+ *  <sl-progress-bar label="Downloading file">
+ *     <span>40% of 100%</span>
+ *  </sl-button-bar>
+ * ```
+ *
+ * @slot default - A place for helper text like e.g. `20% of 100%`.
+ */
+@localized()
+export class ProgressBar extends ScopedElementsMixin(LitElement) {
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-icon': Icon
+    };
+  }
+
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /** Progress value (from 0...100). */
+  @property({ type: Number }) value = 0;
+
+  /** Whether the progress bar has the indeterminate state. */
+  @property({ type: Boolean, reflect: true }) indeterminate = false;
+
+  /** Label describing the value of the progress bar. */
+  @property() label?: string;
+
+  /** The state of the progress bar. */
+  @property({ reflect: true }) state: ProgressState = 'active';
```

Comment:
If the default state is `'active'`, why not make that `undefined` instead? Then you could simply have it as default styling within CSS as well.

Reply:
I prefer to leave the `active` here if it's ok. It's used for example here:
<img width="730" alt="image" src="https://github.com/sl-design-system/components/assets/111562742/f227d81d-bb4e-4500-89ad-e6b2137ad831">


and also for the aria-live with translations etc.
<img width="681" alt="image" src="https://github.com/sl-design-system/components/assets/111562742/54bb4d0e-c34c-458b-b33b-88b367d60781">

Reply:
For your first question, you could simply make those the default.
For the second: i'm not really sure what `${msg('state')} ${msg(this.state)}` is supposed to do?

Also, what happens if somebody does `state=""` or `.state=${undefined}`?

Reply:
For the second - This is used to inform the user via screen reader about state changes, the default is `active` so there will be no such situation with `undefined` or `""`, but ok I can change it to make it that way with removing `active`. :)

Reply:
Changed :)